File: app/dbmodels/CartridgeItem.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'CartridgeItem',
    'menu'          => 'Asset'
);
$table['fields'] = array(
    'id'                    => DBModels::type('increments',
                                              array('visible' => false)),
    'name'                  => DBModels::type('string'),
    'is_deleted'            => DBModels::type('boolean',
                                              array('visible' => false)),
    'manufacturer_id'       => DBModels::type('integer'),
    'comment'               => DBModels::type('text'),
    'location_id'           => DBModels::type('integer'),
    'user_tech_id'          => DBModels::type('integer'),
    'group_tech_id'         => DBModels::type('integer'),
    'cartridge_item_type_id' => DBModels::type('integer'),
    'alarm_threshold'       => DBModels::type('integer', array('defaut'=>10)),
);

$table['relationships'] = array(
    'cartridge_item_type' => array(
        'type' => 'belongsTo',
        'item' => 'CartridgeItemType'),
    'cartridges'   => array(
        'type'  => 'hasMany',
        'item'  => 'Cartridge',),
    'contracts' => array(
        'type'      => 'morphToMany',
        'item'      => 'Contract',
        'table'     => 'linked_contracts'),
    'documents' => array(
        'type'      => 'morphToMany',
        'item'      => 'Document',
        'table'     => 'linked_documents'),
    'infocom' => array(
        'type' => 'morphMany',
        'item' => 'Infocom'),
    'group_tech' => array(
        'type'  => 'belongsTo',
        'item'  => 'Group'),
    'location' => array(
        'type' => 'belongsTo',
        'item' => 'Location'),
    'manufacturer' => array(
        'type' => 'belongsTo',
        'item' => 'Manufacturer'),
    'printermodels' => array(
        'type'      => 'belongsToMany',
        'item'      => 'PrinterModel',
        'linktable' => 'cartridge_compatibilities'),
    'user_tech' => array(
        'type'  => 'belongsTo',
        'item'  => 'User'),
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/cartridge_item.py
from fusionglpi import db

printer_models = db.Table('cartridge_compatibility',
                          db.Column('cartridge_item_id', db.Integer,
                                    db.ForeignKey('cartridge_item.id')),
                          db.Column('printer_model_id', db.Integer,
                                    db.ForeignKey('printer_model.id'))
                          )


class CartridgeItem(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, nullable=False, server_default="0")
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    manufacturer_id = db.Column(db.Integer, nullable=False,
                                server_default="0")
    comment = db.Column(db.Text)
    location_id = db.Column(db.Integer, nullable=False, server_default="0")
    user_tech_id = db.Column(db.Integer, nullable=False, server_default="0")
    group_tech_id = db.Column(db.Integer, nullable=False, server_default="0")
    cartridge_item_type_id = db.Column(db.Integer, nullable=False,
                                       server_default="0")
    alarm_threshold = db.Column(db.Integer, nullable=False,
                                server_default="10")

    cartridges = db.relationship('Cartridge', backref='cartridge_item')
    printer_models = db.relationship('PrinterModel',
                                     secondary=printer_models,
                                     backref=db.backref('cartridge_item',
                                                        lazy='dynamic'))
